forwards_paper.tex

```diff
@@ -306,8 +320,26 @@ \subsection*{Recording the pedigree in forwards time}
 $\randomuniform(A)$ to denote an element of the set $A$ chosen uniformly at random 
 (and each such instance represents and independent draw).
 
+We begin in~\algref{W1} by allocating our initial population $P$
+(a vector of $N$ node IDs),
+and creating $N$ nodes with birth time $T$ generations ago, recorded in the
+vector $\tau$. The set $E$ is used to store the edges that we output during the
+simulation, and $n$ is the number of nodes created so far (and so, the ID of
+the next node we create). Steps~\algref{W2} and~\algref{W3} simply loop over
+the generation clock $t$ and individual index $j$. In~\algref{W4} we check if
+an individual $P_j$ has died in this generation. If it has, we replace it in
+steps \algref{W5}--\algref{W7}; if not, we proceed immediately to the next
+individual. When an individual in the population dies, we first allocate
+a new node with ID $n$ in~\algref{W5} and record its birth time. Then,
+in step~\algref{W6} we choose two indexes $a$ and $b$ uniformly (giving us
+parental IDs $P_a$ and $P_b$) and choose a breakpoint $x$. We record the effects of this
+event by storing two new edges: one recording that the parent of node $n$
+from $0$ to $x$ is $P_a$, and another recording that the parent of $n$
+from $x$ to $L$ is $P_b$. We then complete the replacement event by incrementing
+$n$, ready to represent the next new node.
+
```

Comment:
Ok thanks, this clears things up. If previewing the alg is not appealing to some, I'd say adding just the description:

> our initial population $P$
(a vector of $N$ node IDs),
and creating $N$ nodes with birth time $T$ generations ago, recorded in the
vector $\tau$. The set $E$ is used to store the edges that we output during the
simulation, and $n$ is the number of nodes created so far (and so, the ID of
the next node we create) 

prior to alg W would help readers understand the population state